docs/design/coreclr/botr/vectors-and-intrinsics.md

```diff
@@ -0,0 +1,189 @@
+- The variable sized `Vector<T>`. This represents vector data of indeterminate size. In any given process the size of a `Vector<T>` is the same in all methods, but this size may differ between various machines or environment variable settings read at startup of the process. The `T` type variable may be any of the primitive types, and allows use of integer or double data within a vector. The size and alignment of `Vector<T>` is unknown to the developer, and `Vector<T>` may not exist in any interop signature. Operations on these vector types are supported on all architectures and platforms, although some architectures may optimize various operations.
+- `Vector64<T>`, `Vector128<T>`, and `Vector256<T>` represent sized vectors that closely resemble the fixed size vectors available in C++. These structures can be used in any code that runs, but very few features are supported. These vector types define very few direct features, but instead most operations other than creation use the processor specific hardware intrinsics apis.
+- Processor specific hardware intrinsics apis such as `System.Runtime.Intrinsics.X86.Ssse3`. These apis map directly to individual instructions or short instruction sequences that are specific to a particular hardware instruction. These apis are only useable on hardware that supports the particular instruction. See https://github.com/dotnet/designs/blob/master/accepted/2018/platform-intrinsics.md for the design of these.
+
+# How to use intrinsics apis
+
+There are 3 models for use of intrinsics apis.
+
+1. Usage of `Vector2`, `Vector3`, `Vector4`, and `Vector<T>`. For these, its always safe to just use the types. The jit will generate code that is as optimal as it can for the logic, and will do so unconditionally.
+2. Usage of `Vector64<T>`, `Vector128<T>`, and `Vector256<T>`. These types may be used unconditionally, but are only truly useful when also using the platform specific hardware intrinsics apis.
+3. Usage of platform intrinsics apis. All usage of these apis should be wrapped in an `IsSupported` check of the appropriate kind. Then, within the `IsSupported` check, other platform specific apis may be used.
+
+# Effect of usage of hardware intrinsics on how code is generated
+
+Hardware intrinsics have dramatic impacts on codegen, and the codegen of these hardware intrinsics is dependent on what knowledge of the target machine is available when the code is compiled.
+
+If the code is compiled at runtime by the JIT in a just-in-time manner, then the JIT will generate the best code it can for a given intrinsic usage when compiling tier 1 code, and generate functionally equivalent code when compiling tier 0 code. MethodImplOptions.AggressiveOptimization may be used to disable compilation of tier 0 code.
```

Comment:
```suggestion
If the code is compiled at runtime by the JIT in a just-in-time manner, then the JIT will generate the best code it can based on the current processor's ISA when compiling tier 1 code, and generate functionally equivalent code when compiling tier 0 code. MethodImplOptions.AggressiveOptimization may be used to bypass compilation of tier 0 code and always produce tier 1 code for the method.
```